### PR TITLE
Wildcard matching edge-case correctness fixes

### LINKED ISSUE #321: Wildcard matching edge-case correctness fixes
## Summary
Fix wildcard matching correctness for domain normalization edge cases.

## Scope
- Audit matcher behavior for punycode, dots, and subdomain boundaries.
- Align wildcard matching semantics across runtime and diagnostics.
- Add focused tests for previously failing domain patterns.

## Acceptance criteria
- Wildcard matching returns consistent results for edge-case domains.
- Runtime blocking behavior matches preview/diagnostics outputs exactly.

### PR DESCRIPTION
**Pull request by utilForever**
## What

Improve wildcard/domain matching correctness for dotted and punycode-style edge cases, and ensure runtime blocking and CLI diagnostics compute effective blocked sites through the same shared logic.

## Why

Issue #321 called out correctness drift risk around wildcard matching semantics across normalization, runtime behavior, and diagnostics output. This change removes duplicated matching code paths and adds targeted regressions so the same domain rules produce the same results everywhere.

Closes #321.

### Required

- [ ] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [ ] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #25`)

### Functional Validation

- [ ] Pomodoro timer behavior was verified for this change (if applicable)
- [x] Site blocking behavior was verified for this change (if applicable)
- [ ] WakaTime integration behavior was verified for this change (if applicable)
- [x] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [ ] User-facing docs were updated (`README.md` or relevant docs, if applicable)
- [ ] New dependencies/configuration are documented (if applicable)
- [x] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed)
- [ ] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved blocklist resolution and canonicalization: leading/trailing dots, punycode, and wildcard forms are handled consistently; allowlist entries correctly override matching blocked patterns and results are deduplicated case-insensitively.

* **Tests**
  * Added tests validating canonicalization, wildcard/allowlist interaction, and blocking behavior across dotted and punycode variants.

[Review Change Stack](https://app.coderabbit.ai/change-stack/utilForever/focustime/pull/352?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -602,6 +603,67 @@ impl Default for BlocklistCategoryConfig {
...
+fn block_rule_excluded_by_allowlist(block_rule: &str, allowlist_rules: &[String]) -> bool {
+    if block_rule.starts_with("*.") {
+        return allowlist_rules
+            .iter()
+            .any(|allow_rule| allow_rule.eq_ignore_ascii_case(block_rule));
+    }
+    allowlist_rules
+        .iter()
+        .any(|allow_rule| domain_rule_matches_host(allow_rule, block_rule))
+}
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/config.rs:638]: Implemented. I updated wildcard allowlist handling so broader wildcard allowlist rules cover narrower wildcard block rules (for example, `*.example.com` now excludes `*.api.example.com`), and added focused config regression tests for both wildcard coverage and exact-host non-coverage cases.

File: src/cli/output.rs [hunk 2]
```diff
@@ -1022,4 +964,22 @@ mod tests {
...
+                "api.example.com.".to_string(),
+            ],
+            allowlist_sites: vec![".example.com".to_string()],
+            ..crate::config::BlocklistProfileConfig::default()
+        };
+
+        assert_eq!(
+            effective_blocked_sites_for_profile(&profile),
+            vec!["*.xn--bcher-kva.example".to_string()]
+        );
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/cli/output.rs:983]: Thanks for the review. I am not applying this one because this PR intentionally canonicalizes leading-dot rules as wildcard rules (`.example.com` -> `*.example.com`) based on the agreed issue scope, so `api.example.com` is expected to be excluded by that allowlist in this regression case.
